### PR TITLE
add send_vectored for unix socket

### LINKED ISSUE #68612: UnixDatagram does not support vectored send
The `UnixDatagram` type provides `send` and `send_to` methods, but does not provide a vectored send operation. Such an operation is particularly helpful for datagrams, where separate send operations result in separate datagrams; a vectored send allows sending data from multiple buffers in a single datagram.

Given that `UnixDatagram` only runs on UNIX, where `writev` works just fine on a socket, such an operation could use `writev`; alternatively, this could use `sendmsg`, which supports supplying an iovec. `sendmsg` would also allow a vectored `send_to` operation.

### PR DESCRIPTION
**Pull request by contrun**
closes https://github.com/rust-lang/rust/issues/68612

Currently only supports `send` as I didn't find binding for `sendmsg`. `connect` and `send` costs two syscalls.

### CODE REVIEW

File: src/libstd/sys/unix/ext/net.rs [hunk 1]
```diff
@@ -1408,6 +1408,32 @@ impl UnixDatagram {
...
+    ///
+    /// fn main() -> std::io::Result<()> {
+    ///     let sock = UnixDatagram::unbound()?;
+    ///     sock.connect("/some/sock").expect("Couldn't connect");
+    ///     let bufs = [IoSlice::new(b" "), IoSlice::new(b"hello world")];
+    ///     sock.send_vectored(&bufs).expect("send_vectored function failed");
+    ///     Ok(())
+    /// }
+    /// ```
+    #[stable(feature = "unix_socket", since = "1.10.0")]
```
Comment (contrun) [inline src/libstd/sys/unix/ext/net.rs]: What is the correct way to specify this attribute? I didn't find the relevant documentation.

**Comment by LinkTed** [comment]
I wrote a similar functions in PR #69864.

**Comment by joshtriplett** [comment]
@contrun As far as I can tell, the changes that @LinkTed made in https://github.com/rust-lang/rust/pull/69864 do use `libc::sendmsg`. I think it may make sense to use a common implementation here.

**Comment by contrun** [comment]
@LinkTed @joshtriplett Thanks. I will close this one.

**Comment by Ralith** [comment]
It might be nice to get this in separately from #69864, since just `send_vectored` doesn't require working out an API for ancillary data and can presumably be reviewed and merged much more quickly.

**Pull request closed without merging**